### PR TITLE
INT-4034 - Support for ingesting Snyk projects

### PR DESCRIPTION
**Pull request by austinkelleher**
| Resources    | Entity `_type` | Entity `_class` |
| ------------ | -------------- | --------------- |
| Snyk Project | `snyk_project` | `Project`       |

### CODE REVIEW

File: src/constants.ts [hunk 1]
```diff
@@ -27,8 +28,13 @@ export const Entities = {
   },
   SNYK_FINDING: {
     _type: 'snyk_finding',
-    _class: 'Finding',
-    resourceName: 'Snyk Issue',
+    _class: 'finding',
+    resourceName: 'snyk issue',
+  },
```
Comment (zemberdotnet) [inline src/constants.ts:33]: Did you mean to change line 31 & 32? Also for comparability with new SDK like 31 should probably be:
```
 _class: ["Finding"]
 ```

Comment (austinkelleher) [reply src/constants.ts:33]: Updated. Thanks!

File: docs/jupiterone.md [hunk 2]
```diff
@@ -88,14 +88,16 @@ The following entities are created:
 | CVE          | `cve`          | `Vulnerability`      |
 | CWE          | `cwe`          | `Weakness`           |
 | Snyk Account | `snyk_account` | `Service`, `Account` |
-| Snyk Issue   | `snyk_finding` | `Finding`            |
+| Snyk Project | `snyk_project` | `Project`            |
+| snyk issue   | `snyk_finding` | `Finding`            |
```
Comment (VDubber) [inline docs/jupiterone.md:92]: nit: `Snyk Issue`

File: src/client.ts [hunk 3]
```diff
@@ -104,27 +104,27 @@ export class APIClient {
    * @param iteratee receives each resource and produces entities/relationships
    */
   public async iterateIssues(
-    project: Project,
+    projectId: string,
```
Comment (VDubber) [inline src/client.ts:107]: 👏

File: src/steps/fetchFindings.ts [hunk 4]
```diff
@@ -71,64 +37,70 @@ async function fetchFindings({
...
+          }
+        } else {
+          finding.targets.push(projectSourceName);
+          finding.identifiedInFile = packageName;
+          entityCache.findingEntities[finding.id] = finding;
+
+          for (const cve of finding.cve || []) {
+            let cveEntity = entityCache.cveEntities[cve];
+            if (!cveEntity) {
+              cveEntity = createCVEEntity(cve, issue.issueData.cvssScore!);
```
Comment (VDubber) [inline src/steps/fetchFindings.ts:71]: This looks solid. cve is not added to jobState

File: package.json [hunk 5]
```diff
@@ -19,19 +19,20 @@
...
   },
   "peerDependencies": {
-    "@jupiterone/integration-sdk-core": "^6.19.0"
+    "@jupiterone/integration-sdk-core": "^8.13.11"
   },
   "devDependencies": {
-    "@jupiterone/integration-sdk-core": "^6.19.0",
-    "@jupiterone/integration-sdk-dev-tools": "^6.19.0",
-    "@jupiterone/integration-sdk-testing": "^6.19.0",
+    "@jupiterone/integration-sdk-core": "^8.13.11",
```
Comment (VDubber) [inline package.json:33]: 👍

**Review by VDubber — APPROVED** [review]
Minor comments added. Great changes 👍

File: docs/jupiterone.md (hunk 2 above)
Comment (austinkelleher) [reply docs/jupiterone.md:93]: Thanks addressed

**Review by VDubber — APPROVED** [review]

**Pull request merged**